Sending a query when the origin Prometheus is down panics

```
time=2018-04-02T17:38:39.670700386Z app=trickster caller=trickster/handlers.go:219 level=error event="error downloading url" url="http://localhost:9009/api/v1/query?query=rate%28process_cpu_seconds_total%5B10s%5D%29" detail="Get http://localhost:9009/api/v1/query?query=rate%28process_cpu_seconds_total%5B10s%5D%29: dial tcp 127.0.0.1:9009: connect: connection refused"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x715828]

goroutine 60 [running]:
main.(*TricksterHandler).fetchPromQuery(0xc4200905c0, 0xc420280240, 0x21, 0xc420074d18, 0xc4201e4400, 0x7b4580, 0xc4204fa540, 0x7f8845, 0x4)
        /go/src/github.com/comcast/trickster/handlers.go:311 +0x438
main.(*TricksterHandler).getVectorFromPrometheus(0xc4200905c0, 0xc420280240, 0x21, 0xc420087d18, 0xc4201e4400, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/github.com/comcast/trickster/handlers.go:240 +0xb5
main.(*TricksterHandler).originRangeProxyHandler.func3(0xc420273d40, 0xc4200905c0, 0xc4201d8070, 0xc4202a4720, 0xc420184020, 0xc4201ba7c0, 0xc4202a4710)
        /go/src/github.com/comcast/trickster/handlers.go:649 +0x216
created by main.(*TricksterHandler).originRangeProxyHandler
        /go/src/github.com/comcast/trickster/handlers.go:641 +0x164d
```

Caused by a `nil` response at https://github.com/Comcast/trickster/blob/b9b9b9bcbd5d9f34dc1bf72e2827b250cd3d880b/handlers.go#L311 returned by https://github.com/Comcast/trickster/blob/b9b9b9bcbd5d9f34dc1bf72e2827b250cd3d880b/handlers.go#L217

--- a/handlers.go
+++ b/handlers.go
@@ -255,7 +255,7 @@ func (t *TricksterHandler) getMatrixFromPrometheus(url string, params url.Values
 	// Make the HTTP Request - dont use fetchPromQuery here, that is for instantaneous only.
 	body, resp, duration := t.getURL(hmGet, url, params, getProxyableClientHeaders(r))
 
-	if resp.StatusCode == 200 {
+	if resp != nil && resp.StatusCode == 200 {
 		// Unmarshal the prometheus data into another PrometheusMatrixEnvelope
 		err := json.Unmarshal(body, &pe)
 		if err != nil {
@@ -307,6 +307,10 @@ func (t *TricksterHandler) fetchPromQuery(originURL string, params url.Values, r
 	if err != nil {
 		// Cache Miss, we need to get it from prometheus
 		body, resp, duration = t.getURL(hmGet, originURL, params, getProxyableClientHeaders(r))
+
+		if resp == nil {
+			return body, nil
+		}
 
 		t.Metrics.ProxyRequestDuration.WithLabelValues(originURL, otPrometheus, mnQuery, crKeyMiss, strconv.Itoa(resp.StatusCode)).Observe(float64(duration))
 		t.Cacher.Store(cacheKey, string(body), ttl)
@@ -627,7 +631,7 @@ func (t *TricksterHandler) originRangeProxyHandler(cacheKey string, originRangeR
 					}
 					m.Unlock()
 
-					if r.StatusCode == 200 && udd.Status == rvSuccess {
+					if r != nil && r.StatusCode == 200 && udd.Status == rvSuccess {
 						upperDeltaData = udd
 						t.Metrics.ProxyRequestDuration.WithLabelValues(ctx.Origin.OriginURL, otPrometheus,
 							mnQueryRange, ctx.CacheLookupResult, strconv.Itoa(r.StatusCode)).Observe(float64(duration))
@@ -654,7 +658,7 @@ func (t *TricksterHandler) originRangeProxyHandler(cacheKey string, originRangeR
 					}
 					m.Unlock()
 
-					if r.StatusCode == 200 && ffd.Status == rvSuccess {
+					if r != nil && r.StatusCode == 200 && ffd.Status == rvSuccess {
 						fastForwardData = ffd
 					}
 					wg.Done()

